go/host/db/batches.go

```diff
@@ -132,6 +132,32 @@ func (db *DB) GetBatchBySequenceNumber(sequenceNumber *big.Int) (*common.ExtBatc
 	return db.GetBatch(*batchHash)
 }
 
+// GetBatchListing returns BatchListingResponse given a pagination
+// todo change this when the db changes - this is not super performant
+func (db *DB) GetBatchListing(pagination *common.QueryPagination) (*common.BatchListingResponse, error) {
+	// fetch requested batches
+	var batches []common.PublicBatchListing
+	for i := pagination.Offset; i < pagination.Offset+uint64(pagination.Size); i++ {
+		extBatch, err := db.GetBatchBySequenceNumber(big.NewInt(int64(i)))
```

Comment:
For things like batches I'd expect the pagination to work in reverse somehow?

Like `db.GetBatchBySequenceNumber(headSeqNum - i)` sort of thing to get the latest 20 batches or whatever.

Should be rare that users would go back far back I guess but it's slightly tricky when the latest keeps ticking up I guess.

Reply:
It's gonna take a bit of fe engineering, but you're right it's more intuitive, let me do that in the next round.